core/pkg/net/ssl/ssl.go

```diff
@@ -125,6 +127,56 @@ func AddOrUpdateCertAndKey(name string, cert, key, ca []byte) (*ingress.SSLCert,
 	}, nil
 }
 
+// AddOrUpdateCertAuth creates a .pem file with the specified CAs to be used in Cert Authentication
+func AddOrUpdateCertAuth(name string, ca []byte) (*ingress.SSLCert, error) {
+
+	caName := fmt.Sprintf("ca-%v.pem", name)
+	caFileName := fmt.Sprintf("%v/%v", ingress.DefaultSSLDirectory, caName)
+
+	tempCAPemFile, err := ioutil.TempFile(ingress.DefaultSSLDirectory, caName)
+	glog.V(3).Infof("AddOrUpdateCertAuth: Creating temp file %v for CA: %v", tempCAPemFile.Name(), caName)
+
+	if err != nil {
+		return nil, fmt.Errorf("could not create temp CA pem file %v: %v", tempCAPemFile.Name(), err)
+	}
+
+	_, err = tempCAPemFile.Write(ca)
+	if err != nil {
+		return nil, fmt.Errorf("could not write to CA pem file %v: %v", tempCAPemFile.Name(), err)
+	}
+
+	err = tempCAPemFile.Close()
+	if err != nil {
+		return nil, fmt.Errorf("could not close CA temp pem file %v: %v", tempCAPemFile.Name(), err)
+	}
+
+	pemCACerts, err := ioutil.ReadFile(tempCAPemFile.Name())
```

Comment:
why write it then read it then validate, can you validate the raw CA before even creating the file so we can exit early if it's invalid? i.e something like
```go
// AddCA validates then writes the given CA to a pem file located in the DefaultSSLDirectory (where is this coming from?)
// and returns an SSLCert struct containing the location the CA was written to. 
AddCA(name, ca []bytes) (SSLCert, error) {
  if err := validate(ca) != nil {
    return nil, err
  }
 // write CA to path
 return ingress.SSLCerts{}, err
}
```